getting-started/troubleshooting.html.md

```diff
@@ -1,176 +1,223 @@
-...
+
+fastify.listen({ port: 3000, host: '0.0.0.0' }, function (err, address) {
+  if (err) {
+    fastify.log.error(err)
+    process.exit(1)
+  }
+  fastify.log.info(`server listening on ${address}`)
+})
 ```
 
-And redeploy that, it will work.
+Then make sure that the `internal_port` value in `fly.toml` is set to `3000`. 
 
-## _Inspecting with SSH_
+## Health checks failing
 
-You can use `flyctl ssh console` to connect to a running instance of your application. Use `flyctl ssh console -s` to select a specific instance.
+We don't automatically add health checks to your `fly.toml` file when you cerate your app. The health checks that you subsequently add to your app can fail for a number of reasons.
 
-## _Health checks failing_
+A good first step can be to look at a failed Machine and see what you can figure out. To see the specific Machine status, run `fly status --all` to get a list of Machines in your app. Find one with state `failed` or `stopped`, then run `fly machine status <machine id>` . This will give you a lot more information. Make sure you check the exit code, if it’s 0 it means health check failures, if it’s not zero it’s some issue crashing the process.
 
-If your health checks keep failing during deployment...
+### Out-of-memory (OOM) or high CPU usage
 
-```
-[error] Health check status changed 'warning' => 'critical'
+If your Machine's resources are reaching their limits, then this could slow everything down, including accepting connections and responding to HTTP requests. Slow responses can trigger health check failures.
 
-```
+You might see out-of-memory errors in logs. Some apps (like Node.js apps that use Prisma) can be RAM-intensive. So your app may be killed for out-of-memory (OOM) reasons. The solution is to just [add more RAM](https://fly.io/docs/apps/scale-machine/#add-ram). 
 
-... the first step is to look at a failed VM and see what you can figure out. RAM increases are only useful if the VM had an out of memory error (which you might see in the logs). The health check grace period is only helpful if health checks took too long to pass.
+If you see high CPU usage in metrics you might need to select a new [preset CPU/RAM combination](/docs/apps/scale-machine/#select-a-preset-cpu-ram-combination), or even update only the [CPU on an individual Machine](/docs/apps/scale-machine/#machines-not-belonging-to-fly-launch).
 
-To see the specific VM status, run `fly status --all` to get a list of VMs. Find one with status `failed` , then run `fly vm status <id>` . This will give you a lot more information. Make sure you check the exit code, if it’s 0 it means health check failures, if it’s not zero it’s some issue crashing the process.
+### Grace period
 
-To increase the grace period for your app (ex: if it takes about 30 seconds for your application to boot), you would do so:
-```
-  # If you are using tcp_checks
+Grace period is the time to wait after a Machine starts up before checking its health.
```

Comment:
I don't think this is really what grace period is on Machines unless it's been improved since it was first implemented

Reply:
@catflydotio this came from the [fly.toml docs](https://fly.io/docs/reference/configuration/) but I can look into it

Reply:
So I think this is essentially right. it's the time to wait before letting flyctl make any decisions about healthy vs unhealthy Machines